### PR TITLE
Support IPv6

### PR DESCRIPTION
**Pull request by melle**

### CODE REVIEW

File: UDPBroadcast/UDPv6BroadcastConnection.swift [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        if ret == -1 {
+            debugPrint("Couldn't enable broadcast on socket")
+            close(newSocket)
+            throw ConnectionError.enableBroadcastFailed
+        }
+
+        // Bind socket if needed
+        if shouldBeBound {
+            // FIXME: not yet implemented
+            debugPrint("FIXME: implement IPv6 binding!")
```
Comment (zweigraf) [inline UDPBroadcast/UDPv6BroadcastConnection.swift]: Will this be done?

**Comment by zweigraf** [comment]
If we use this in production, I would love to see some better types on the API of this thing. (IP address not as String, ports not as Int, etc).

**Review by zweigraf — APPROVED** [review]

File: UDPBroadcast/UDPv6BroadcastConnection.swift (hunk 1 above)
Comment (melle) [reply UDPBroadcast/UDPv6BroadcastConnection.swift]: Probably not. This was added to the original project by a 3rd party but we will never use it. I'm tempted to remove this addition to keep the whole thing as simple as possible.

**Comment by melle** [comment]
ok, this breaks compatibility with upstream (removed bindIt, added addressFamily). Higher level types (like Network.framework's IPv6Address) should not be used here as this adds a dependency to iOS 12.

File: UDPBroadcast/UDPBroadcastConnection.swift [hunk 2]
```diff
@@ -150,11 +267,81 @@ open class UDPBroadcastConnection: UDPBroadcastSocketCreating {
...
+        }
     }
     
-    open func sendBroadcast(data: Data) throws {
-        fatalError("Must be implemended in a subclass.")
+    /// Send broadcast data.
+    ///
+    /// - Parameter data: Data to send via broadcast.
+    /// - Throws: Throws a `ConnectionError` if an error occurs.
+    private func sendBroadcastv4(data: Data) throws {
```
Comment (zweigraf) [inline UDPBroadcast/UDPBroadcastConnection.swift:283]: Aren't those two functions here basically the same, apart from `v4address` and `v6address`?

Comment (melle) [reply UDPBroadcast/UDPBroadcastConnection.swift:283]: yes, I tried to untangle this yesterday with Luizb and failed. We can retry to reduce the redundancy later.

**Pull request merged**